src/nbl/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -955,6 +976,49 @@ SContext::shape_ass_type CMitsubaLoader::loadBasicShape(SContext& ctx, uint32_t
 	return mesh;
 }
 
+void CMitsubaLoader::cacheEmissionProfile(SContext& ctx, const CElementEmissionProfile* profile)
+{
+	if (!profile)
+		return;
+	
+	const auto cacheKey = ctx.emissionProfileCacheKey(*profile);
+
+	const asset::IAsset::E_TYPE types[]{ asset::IAsset::ET_IMAGE_VIEW, asset::IAsset::ET_TERMINATING_ZERO };
+	if (ctx.override_->findCachedAsset(cacheKey, types, ctx.inner, 0u).getContents().empty())
+	{
+		auto assetLoaded = interm_getAssetInHierarchy(m_assetMgr, profile->filename, ctx.inner.params, 0u, ctx.override_);
+
+		ICPUImageView::SCreationParams viewParams = {};
+		auto contentRange = assetLoaded.getContents();
+		if (contentRange.empty())
+		{
+			os::Printer::log("[ERROR] Could Not Find Texture: " + cacheKey, ELL_ERROR);
+			return;
+		}
+		viewParams.image = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(*contentRange.begin());
+		viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0);
+		viewParams.viewType = IImageView<ICPUImage>::ET_2D;
+		viewParams.format = viewParams.image->getCreationParameters().format;
+		viewParams.subresourceRange.aspectMask = static_cast<IImage::E_ASPECT_FLAGS>(0);
+		viewParams.subresourceRange.levelCount = viewParams.image->getCreationParameters().mipLevels;
+		viewParams.subresourceRange.layerCount = 1u;
+
+		asset::SAssetBundle viewBundle(nullptr, { ICPUImageView::create(std::move(viewParams)) });
+		ctx.override_->insertAssetIntoCache(std::move(viewBundle), cacheKey, ctx.inner, 0u);
+	}
+	
+	{
+		ISampler::SParams samplerParams = ctx.emissionProfileSamplerParams(*profile);
+		const asset::IAsset::E_TYPE types[] = { asset::IAsset::ET_SAMPLER, asset::IAsset::ET_TERMINATING_ZERO };
+		const std::string samplerCacheKey = ctx.samplerCacheKey(samplerParams);
+		if (ctx.override_->findCachedAsset(samplerCacheKey, types, ctx.inner, 0u).getContents().empty())
+		{
+			SAssetBundle samplerBundle(nullptr, { core::make_smart_refctd_ptr<ICPUSampler>(samplerParams) });
+			ctx.override_->insertAssetIntoCache(std::move(samplerBundle), samplerCacheKey, ctx.inner, 0u);
+		}
+	}
```

Comment:
you won't need this if your IES Loader will already return a Image View, you'll only need to deduce a sampler to use from the metadata